pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +100,21 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	if m.updates == nil {
```

Comment:
I decided I preferred the current API to trying to push this to the constructor, so I made updates thread-safe. It's a bit messy though, so let me know if you have a better idea.